lib/metanorma/plugin/glossarist/dataset_preprocessor.rb

```diff
@@ -109,6 +110,7 @@ def process_line(document, input_lines, current_line, liquid_doc)
           elsif match = current_line.match(GLOSSARIST_BLOCK_REGEX)
             process_glossarist_block(document, liquid_doc, input_lines, match)
           else
+            @title_depth = current_line.sub(/ .*$/, "").size if /^==+ \S/.match?(current_line)
```

Comment:
You cannot do this the way Asciidoctor is set up: scalar class variables in plugins are immutable, because plugin classes are frozen, so this crashes Metanorma:

````
/Users/nickn/.rbenv/versions/3.3.3/lib/ruby/gems/3.3.0/bundler/gems/metanorma-plugin-glossarist-bff0909b129a/lib/metanorma/plugin/glossarist/dataset_preprocessor.rb:113:in `process_line': 
asciidoctor: FAILED: /Users/nickn/Documents/Arbeit/upwork/ribose/metanorma-ogc/a.adoc: Failed to load AsciiDoc document - can't modify frozen Metanorma::Plugin::Glossarist::DatasetPreprocessor: 
#<Metanorma::Plugin::Glossarist::DatasetPreprocessor:0x000000016db3aab0 @config={:file_system=>#<Liquid::LocalFileSystem:0x000000016d8f7aa0 @root="/Users/nickn/Documents/Arbeit/upwork/ribose/metanorma-ogc", @pattern="_%s.liquid">}, @datasets={"dataset"=>Liquid::Drops::ConceptsDrop}, @title_depth=2, @rendered_bibliographies={}> (FrozenError)
````

The workaround I have had to use in the past is to make class variables be objects: this complaint won't be raised if you make `@title_depth` be `{value: 2 }`

Reply:
Got it, will fix and push the changes in few minutes.